deployments/azure-staticwebapp.yml

```diff
@@ -28,8 +28,14 @@ env:
   APP_ARTIFACT_LOCATION: "build" # location of client code build output
   AZURE_STATIC_WEB_APPS_API_TOKEN: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN }} # secret containing deployment token for your static web app
 
+permissions:
+  contents: read
+
 jobs:
   build_and_deploy_job:
+    permissions:
+      contents: read # for actions/checkout to fetch code
+      pull-requests: write # for Azure/static-web-apps-deploy to comment on PRs
```

Comment:
why does it required` pull-requests: write `permission? Also, do we need to specify `contents: read` here as it is already specified at the workflow level and job should inherit from there.

Reply:
@anuragc617 thanks for reviewing.. you can check [this](https://github.com/actions/starter-workflows/pull/1348#discussion_r792063720) conversation for your clearification..